dashboard/app/controllers/followers_controller.rb

```diff
@@ -52,42 +24,32 @@ def remove
 
     authorize! :destroy, f
     f.delete
-    FollowerMailer.student_disassociated_notify_teacher(@teacher, @user).deliver_now if @teacher.email.present?
-    redirect_to root_path, notice: t('teacher.student_teacher_disassociated', teacher_name: @teacher.name, student_name: @user.name)
+    FollowerMailer.student_disassociated_notify_teacher(@teacher, current_user).deliver_now if @teacher.email.present?
+    redirect_to root_path, notice: t('teacher.student_teacher_disassociated', teacher_name: @teacher.name, student_name: current_user.name)
   end
 
   # GET /join/XXXXXX
-  # join a section as a new student
+  # if logged in, join the section, if not logged in, present a form to create a new user and log in
   def student_user_new
-    @section = Section.find_by_code(params[:section_code])
+    load_section
 
     # make sure section_code is in the path (rather than just query string)
     if request.path != student_user_new_path(section_code: params[:section_code])
       redirect_to student_user_new_path(section_code: params[:section_code])
     elsif current_user && @section
-      if current_user == @section.user
-        redirect_to root_path, alert: I18n.t('follower.error.cant_join_own_section') and return
-      end
+      @section.add_student(current_user)
 
-      follower_same_user_teacher = current_user.followeds.where(:user_id => @section.user_id).first
-      if follower_same_user_teacher.present?
-        follower_same_user_teacher.update_attributes!(:section_id => @section.id)
-      else
-        Follower.create!(user_id: @section.user_id, student_user: current_user, section: @section)
-      end
-
-      current_user.assign_script(@section.script) if @section.script
-
-      redirect_to root_path, notice: I18n.t('follower.registered', section_name: @section.name) and return
+      redirect_to root_path, notice: I18n.t('follower.registered', section_name: @section.name)
+    else
+      @user = User.new
+      # render student_user_new
```

Comment:
oh, yeah, that was intended to be an informative comment but I see how it looks like commented out code